app/src/sharedTest/java/org/oppia/android/app/parser/CustomBulletSpanTest.kt

```diff
@@ -0,0 +1,138 @@
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.CustomBulletSpan
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = CustomBulletSpanTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class CustomBulletSpanTest {
+
+  @Inject
+  lateinit var context: Context
+
+  @Test
+  fun customBulletSpan_checkLeadingMargin() {
+    val context = ApplicationProvider.getApplicationContext<TestApplication>()
+    val bulletRadius = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.bullet_radius
+    )
+    val spacingBeforeBullet = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_bullet
+    )
+    val spacingBeforeText = context.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_text
+    )
+    val expectedMargin = spacingBeforeBullet + spacingBeforeText + 2 * bulletRadius
+    val testString = SpannableString("Text with \nBullet Point")
+    testString.setSpan(BulletSpan(), 10, 22, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    val spannableString = SpannableStringBuilder(testString)
+    val customBulletSpannble = CustomBulletSpan.replaceBulletSpan(spannableString, context)
+    customBulletSpannble.getSpans<CustomBulletSpan>(
+      0,
+      spannableString.length,
+      CustomBulletSpan::class.java
+    ).forEach {
+      val margin = it.getLeadingMargin(true)
+      assertThat(margin).isEqualTo(expectedMargin)
+    }
+    assertThat(customBulletSpannble.getSpans<CustomBulletSpan>().size).isGreaterThan(0)
```

Comment:
Let's split this into a couple of tests (you can move the SpannableString construction to a val or getter to share it). Specifically:
- One test to verify replaceBulletSpan for a string without bullet spans (shouldn't change anything)
- One test to verify replaceBulletSpan with a string containing one bullet span (verify it's replaced)
- One test to verify replaceBulletSpan with multiple bullet spans & a non-bullet span (verify all & only bullet spans are replaced)
- One test to verify replaceBulletSpan for a string already containing custom bullet spans--shouldn't replace them
- A test to verify that leading span is computed correctly

This will really help further improve our behavioral test coverage for this class.

Reply:
Done